### ISSUE TITLE
can't install texlive packages with tlmgr

### ISSUE THREAD

**Issue by kurokirasama** [termux/termux-packages#2262]
Hello, i changed phone so i'm setting up termux. 
I installed texlive and updated tlmgr and tried to install a few packages but i got the following error:

```
Can't locate TeXLive/TLConfig.pm in @INC (you may need to install the TeXLive::TLConfig module) (@INC contains: /data/data/com.termux/files/texmf-dist/scripts/texlive /data/data/com.termux/files/tlpkg /data/data/com.termux/files/usr/lib/perl5/site_perl/5.26.1/aarch64-android /data/data/com.termux/files/usr/lib/perl5/site_perl/5.26.1 /data/data/com.termux/files/usr/lib/perl5/5.26.1/aarch64-android /data/data/com.termux/files/usr/lib/perl5/5.26.1 .) at /data/data/com.termux/files/usr/bin/tlmgr.ln line 99.
BEGIN failed--compilation aborted at /data/data/com.termux/files/usr/bin/tlmgr.ln line 99.
```
Please help!!
My phone, Huawei P9 Lite 2017, Android 7.0

**Comment by Grimler91** [comment]
Have you installed both texlive-bin and texlive or only texlive-bin? Are all packages up to date?

**Comment by kurokirasama** [comment]
yes, both. And yes.

**Comment by Grimler91** [comment]
Weird. 
What's the output of:
ˋkpsewhich --var-value TEXMFROOTˋ
ˋhead $PREFIX/share/texlive/texmf-dist/scripts/texlive/tlmgr.plˋ
ˋhead -n 91 $PREFIX/share/texlive/texmf-dist/scripts/texlive/tlmgr.pl | tail -n 4ˋ

**Comment by kurokirasama** [comment]
In order:

```
/data/data/com.termux/files

#!/data/data/com.termux/files/usr/bin/env perl
# $Id: tlmgr.pl 46207 2018-01-04 18:34:36Z karl $
#
# Copyright 2008-2017 Norbert Preining
# This file is licensed under the GNU General Public License version 2
# or any later version.
#

my $svnrev = '$Revision: 46207 $';
my $datrev = '$Date: 2018-01-04 19:34:36 +0100 (Thu, 04 Jan 2018) $';

# make Perl find our packages first:
  unshift (@INC, "$Master/tlpkg");
  unshift (@INC, "$Master/texmf-dist/scripts/texlive");
}
```

**Comment by Grimler91** [comment]
TEXMFROOT is set to the wrong location. 
Maybe you have some configuration file somewhere messing it up. 

Check what it is set to in the conf file with `head -n 60 $PREFIX/share/texlive/texmf-dist/web2c/texmf.cnf | tail -n 7`
It should say `TEXMFROOT = $SELFAUTODIR/share/texlive`. 
If it is correct then you could try adding `export TEXMFROOT=$PREFIX/share/texlive` to your .bashrc to override whatever file is overriding the variable value.

**Comment by kurokirasama** [comment]
this is the output of your last suggestion, `TEXMFROOT` doesn't appear here:
```
% Redistributors will probably want $SELFAUTODIR/share, i.e., /usr/share.
% Kpathsea sets (in kpathsea/progname.c):
% SELFAUTOLOC (directory of the binary=/usr/local/texlive/YYYY/bin/i386-linux),
% SELFAUTODIR (its parent = /usr/local/texlive/YYYY/bin), 
% SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
% SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
% Sorry for the off-by-one-generation names.
```

**Comment by Grimler91** [comment]
@kurokirasama what's the output of `apt show -a texlive-bin`? 

That texmf.cnf file doesn't come from termux' texlive. I have no idea where you got it from and suggest that you reinstall texlive.

**Comment by kurokirasama** [comment]
```
Package: texlive-bin
Version: 20170524-8
Maintainer: Henrik Grimler @Grimler91
Installed-Size: 27.8 MB
Depends: freetype, libpng, libgd, libgmp, libmpfr, libicu, liblua, poppler, libgraphite, harfbuzz-icu, teckit
Recommends: texlive
Breaks: texlive (<< 20170524-3)
Replaces: texlive (<< 20170524-3)
Homepage: https://www.tug.org/texlive/
Download-Size: 5657 kB
APT-Manual-Installed: yes
APT-Sources: https://termux.net stable/main aarch64 Packages
Description: TeX Live is a distribution of the TeX typesetting system. This package contains architecture dependent binaries.
```

**Comment by Grimler91** [comment]
Ah okay, texmf.cnf was updated in January and running `tlmgr update --all` changes it. 

Thanks for reporting this.

You can edit the texmf.cnf file and change the line `TEXMFROOT = $SELFAUTOPARENT` to `TEXMFROOT = $SELFAUTODIR/share/texlive` to fix it for now.

**Comment by kurokirasama** [comment]
thanks!! it worked.

**Comment by Grimler91** [comment]
Solved in #2392.
